packages/pages/src/vite-plugin/build/closeBundle/functionMetadata.ts

```diff
@@ -26,38 +26,55 @@ const getFunctionMetadataMap = async (): Promise<
 > => {
   const filepaths = glob
     .sync(path.join(FUNCTIONS_PATH, "**/*.{js,ts}"), { nodir: true })
-    .map((f) => path.resolve(f));
-  const functionMetadataArray: [string, FunctionMetadata][] = await Promise.all(
-    filepaths.map(async (filepath) => {
-      const buildResult = await esbuild.build({
-        entryPoints: [filepath],
-        outdir: TEMP_DIR,
-        write: false,
-        format: "esm",
-        bundle: true,
-      });
-      const importedFile = await importFromString(
-        buildResult.outputFiles[0].text
-      );
-      const relativePath = path.relative(process.cwd(), filepath);
-      return [relativePath, { entrypoint: importedFile.default?.name }];
-    })
-  );
+    .map(f => path.resolve(f))
+  const functionMetadataArray: [string, FunctionMetadata][] = []
+
+  await Promise.allSettled(filepaths.map(generateFunctionMetadata)).then(
+    results =>
+      results.forEach(result => {
+        if (result.status === "fulfilled") {
+          functionMetadataArray.push(result.value)
+        } else {
+          console.error(result.reason)
+        }
+      })
+  )
+
   return Object.fromEntries(
     functionMetadataArray.filter(([, metadata]) => metadata.entrypoint)
-  );
-};
+  )
+}
+
+async function generateFunctionMetadata(
+  filepath: string
+): Promise<[string, FunctionMetadata]> {
+  const buildResult = await esbuild.build({
+    entryPoints: [filepath],
+    outdir: TEMP_DIR,
+    write: false,
+    format: "esm",
+    bundle: true,
+  })
+  const importedFile = await importFromString(buildResult.outputFiles[0].text)
+
+  if (!importedFile.default) {
+    return Promise.reject("Blah")
```

Comment:
I believe you can `throw` here and it'll be equivalent behavior (and also a little more async-awaitey)